src/overlays/actors/ovl_En_Sth/z_en_sth.c

```diff
@@ -40,45 +55,357 @@ static ColliderCylinderInit D_80B0B404 = {
+        func_80038290(globalCtx, &this->actor, &this->headRot, &this->unk_2AC, this->actor.focus.pos);
+    } else {
+        Math_SmoothStepToS(&this->headRot.x, 0, 6, 0x1838, 0x64);
+        Math_SmoothStepToS(&this->headRot.y, 0, 6, 0x1838, 0x64);
+        Math_SmoothStepToS(&this->unk_2AC.x, 0, 6, 0x1838, 0x64);
+        Math_SmoothStepToS(&this->unk_2AC.y, 0, 6, 0x1838, 0x64);
+    }
+}
+
+void EnSth_RewardObtainedTalk(EnSth* this, GlobalContext* globalCtx) {
+    if (func_8002F334(&this->actor, globalCtx)) {
+        if (this->actor.params == 0) {
+            EnSth_SetupAction(this, EnSth_ParentRewardObtainedWait);
+        } else {
+            EnSth_SetupAction(this, EnSth_ChildRewardObtainedWait);
+        }
+    }
+    EnSth_FacePlayer(this, globalCtx);
+}
+
+void EnSth_ParentRewardObtainedWait(EnSth* this, GlobalContext* globalCtx) {
+    if (func_8002F194(&this->actor, globalCtx)) {
+        EnSth_SetupAction(this, EnSth_RewardObtainedTalk);
+    } else {
+        this->actor.textId = 0x23; // "We'll be careful not to get cursed again!"
+        if (this->actor.xzDistToPlayer < 100.0f) {
+            func_8002F2CC(&this->actor, globalCtx, 100.0f);
+        }
+    }
+    EnSth_LookAtPlayer(this, globalCtx);
+}
+
+void EnSth_GivePlayerItem(EnSth* this, GlobalContext* globalCtx) {
+    u16 getItemId = sGetItemIds[this->actor.params];
+
+    switch (this->actor.params) {
+        case 1:
+        case 3:
+            switch (CUR_UPG_VALUE(UPG_WALLET)) {
+                case 0:
+                    getItemId = GI_WALLET_ADULT;
+                    break;
+
+                case 1:
+                    getItemId = GI_WALLET_GIANT;
+                    break;
+            }
+            break;
+    }
+
+    func_8002F434(&this->actor, globalCtx, getItemId, 10000.0f, 50.0f);
+}
+
+void EnSth_GiveReward(EnSth* this, GlobalContext* globalCtx) {
+    if (Actor_HasParent(&this->actor, globalCtx)) {
+        this->actor.parent = NULL;
+        EnSth_SetupAction(this, EnSth_RewardObtainedTalk);
+        gSaveContext.eventChkInf[13] |= this->eventFlag;
+    } else {
+        EnSth_GivePlayerItem(this, globalCtx);
+    }
+    EnSth_FacePlayer(this, globalCtx);
+}
+
+void EnSth_RewardUnobtainedTalk(EnSth* this, GlobalContext* globalCtx) {
+    if ((func_8010BDBC(&globalCtx->msgCtx) == 5) && func_80106BC8(globalCtx)) {
+        func_80106CCC(globalCtx);
+        EnSth_SetupAction(this, EnSth_GiveReward);
+        EnSth_GivePlayerItem(this, globalCtx);
+    }
+    EnSth_FacePlayer(this, globalCtx);
+}
+
+void EnSth_RewardUnobtainedWait(EnSth* this, GlobalContext* globalCtx) {
+    if (func_8002F194(&this->actor, globalCtx)) {
+        EnSth_SetupAction(this, EnSth_RewardUnobtainedTalk);
+    } else {
+        if (this->actor.params == 0) {
+            this->actor.textId = 0x28; // "Since you've destroyed [gsTokens] Spiders ..."
+        } else {
+            this->actor.textId = 0x21; // "The curse has been broken! ..."
+        }
```

Comment:
possible ternary? I suppose that makes the comments not really work.

Reply:
luckily the ternary doesnt match anyway :) lol